pkg/stanza/fileconsumer/config.go

```diff
@@ -165,6 +182,12 @@ func (c Config) validate() error {
 		return fmt.Errorf("`fingerprint_size` must be at least %d bytes", MinFingerprintSize)
 	}
 
+	if !featuregate.GetRegistry().IsEnabled(allowFileDeletion) {
+		if c.DeleteAfterRead && c.StartAt == "end" {
+			return fmt.Errorf("`delete_after_read` cannot be used with `start_at: end`")
+		}
+	}
```

Comment:
I think this condition is inverted (this error is only relevant if allowFileDeletion flag is enabled, not if it's disabled)

I also think it's valuable to add the error back in for the case where the feature gate is disabled, but DeleteAfterRead is true. Otherwise, a user could get confused if they skimmed the docs and enabled this option without enabling the featuregate, which would end up just ignoring the option.

Reply:
All good points. I reworked this again. Now just failing fast if trying to use the feature without the gate enabled. Later, assuming there does not need to be a concern about the gate any longer.